gateway/src/resty/balancer.lua

```diff
@@ -87,7 +87,7 @@ function _M.select_peer(self, peers)
   return peer
 end
 
-function _M.set_peer(self, peers)
+function _M.set_peer(self, peers, default_port)
   local balancer = self.balancer
```

Comment:
I’m not really convinced by the API. 

Shouldn’t this function extract the default port?

Reply:
Extract from where? :)

Reply:
I think the best approach is to not use `set_peer` but rather `select_peer` + `balancer.set_current_peer`.